### PR TITLE
added subscribe blocks example

### PR DESCRIPTION
**Pull request by gilescope**
It retrieves me blocks and the parent hash matches the parent hash so LGTM.

### CODE REVIEW

File: examples/examples/rpc_call_subscribe_blocks.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let api = ClientBuilder::new()
+        .set_url("wss://rpc.polkadot.io:443")
+        .build()
+        .await?
+        .to_runtime_api::<polkadot::RuntimeApi<DefaultConfig, PolkadotExtrinsicParams<DefaultConfig>>>();
+
+    // For non-finalised blocks use `.subscribe_finalized_blocks()`
```
Comment (lexnv) [inline examples/examples/rpc_call_subscribe_blocks.rs:49]:
```suggestion
    // For non-finalised blocks use `.subscribe_blocks()`
```

**Comment by jsdw** [comment]
Prolly needs a `cargo +nightly fmt` to be run to fix the CI error.

I'm happy with this for now, at least until we have a "proper" interface for subscribing to these things (I generally consider the RPC interface to be a bit secondary I guess).

File: examples/examples/rpc_call_subscribe_blocks.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    sp_runtime::{
+        generic::Header,
+        traits::BlakeTwo256,
+    },
+    ClientBuilder,
+    DefaultConfig,
+    PolkadotExtrinsicParams,
+};
+
+#[subxt::subxt(runtime_metadata_path = "examples/polkadot_metadata.scale")]
```
Comment (jsdw) [inline examples/examples/rpc_call_subscribe_blocks.rs:36]: The scale metadata moved (see other examples); fixing this should make things compile again :)

**Review by jsdw — APPROVED** [review]
I'm happy with this once it's compiling etc :)

**Review by lexnv — APPROVED** [review]

**Pull request merged**